### ISSUE TITLE
storeAccessToken breaks login after 0.7.0

### ISSUE THREAD

**Issue by tomclark** [auth0/nextjs-auth0#128]
### Description

I'm having an issue where if I add `storeAccessToken` to the session configuration, it results in the session not being created at all in versions of the plugin later than 0.7.0. My application goes through the login sequence with Auth0, is issued a token (I can see this happening successfully in the Auth0 logs), but when Auth0 redirects back to the app, there is no session at all and I'm not logged in.

This only seems to be an issue on versions later than 0.7.0. If I use 0.7.0, it works as expected and the access token is stored in the session.

It's only the access token that causes this to fail. If I include the ID token, the session is created and I can see the ID token in the state, but adding the access token breaks login completely.

I've tried using both a regular web application and a machine-to-machine application, and I've also tried adding some of the legacy grant types:

![image](https://user-images.githubusercontent.com/784012/83683572-866f0680-a5dd-11ea-9bea-ac9ca5cb4048.png)

They're unticked in this screenshot, but I've tried it with all of them ticked as well with the same failed response.

I've tried using the code to specifically get an access token, but it fails:

```javascript
const tokenCache = await auth0.tokenCache(req, res);
const { accessToken } = await tokenCache.getAccessToken();
```

This is no doubt because - as the documentation clearly says - you must mark storeAccessToken true in the settings, but I obviously can't do that since that's what causes the problem. The tokenCache code doesn't appear to be in versions as old as 0.7.0 anyway.

I'm entirely open to the possibility (probability!) that I've not set something up correctly - be it my application in Auth0 or the config of my app or something I'm missing completely - but I'm struggling to see what the issue is, and I'd be very grateful for any assistance from the community. I can obviously continue to use 0.7.0 in the meantime, but it'd be good to upgrade to the latest version.

Thanks very much in advance for any assistance.

### Reproduction

Create a new application using `yarn create next-app --example auth0`, and then upgrade the version of @auth0/nextjs-auth0 to latest (0.13.0), and add storeAccessToken to the module that stores the Auth0 information (auth0.js):

```javascript
import config from './config'
import { initAuth0 } from '@auth0/nextjs-auth0'

export default initAuth0({
  clientId: config.AUTH0_CLIENT_ID,
  clientSecret: config.AUTH0_CLIENT_SECRET,
  scope: config.AUTH0_SCOPE,
  domain: config.AUTH0_DOMAIN,
  redirectUri: config.REDIRECT_URI,
  postLogoutRedirectUri: config.POST_LOGOUT_REDIRECT_URI,
  session: {
    cookieSecret: config.SESSION_COOKIE_SECRET,
    cookieLifetime: config.SESSION_COOKIE_LIFETIME,
    storeIdToken: true,
    storeAccessToken: true, // Removing this line means that login works - but I don't then have an access token to call my API
    storeRefreshToken: true
  }
})
```

### Environment

- **Version of this library used:**

0.7.0 - works as expected.
Later than 0.7.0 fails.

- **Version of the platform or framework used, if applicable:**

NextJS 9.4.4.

I've tried this using Node 12.18.0 and 14.4.0 with the same result.

- **Other relevant versions (language, server software, OS, browser):**

I've tried various browsers: Chrome, Opera, and Safari.

- **Other modules/plugins/libraries that might be involved:**

I can't think of anything significant, but the package.json in the clean example created through `yarn create next-app` looks like this:

```json
{
  "name": "nextjs-auth0-example",
   "scripts": {
      "dev": "next",
      "build": "next build",
      "start": "next start"
    },
  "author": "",
   "license": "MIT",
  "dependencies": {
    "@auth0/nextjs-auth0": "0.13.0",
    "dotenv": "^8.2.0",
    "next": "^9.4.4",
    "react": "^16.12.0",
    "react-dom": "^16.12.0"
  }
}
```

**Comment by tomclark** [comment]
I think I understand what's going on. I noticed during login while I had the dev tools open that a message flashed up and then disappeared, so I screenshotted it:

![image](https://user-images.githubusercontent.com/784012/83779988-8118c700-a684-11ea-8edd-8b2f6cbafb33.png)

Ignoring the message about the unrecommended use of the openid and profile scopes, it seems that the cookie that sets the session is too big and so it doesn't get set. I was a bit confused by this because the ID token is bigger than the access token, and I was positive I'd tried leaving out the ID token, but sure enough, if I swap around the settings, and tell the app to store the access token and the refresh token in session, while leaving out the ID token, login does indeed work and a session is set correctly. Yay!

I confirmed it by leaving the dev tools open again and preserving the log on the network tab as I went through the login process. When I'm redirected back to /api/callback by Auth0, it tries to do this, which is what fails:

```
Set-Cookie: a0:session=Fe26.2**419b30c4d073490abe78e05784e2186236dea5d7eacc8b1ee24c4c896a745778*yLoJVfx6ZTPwm7MYx0NwxA*Q-Vqz6Cr3Cm9acHT83Cd7z2sxoHtxRV_JTk_d_9MajJjRI2I6JJDOXF2l3PjRyXbrlvlFEmM2CpNkopOeH0O47fLxfi857LKfgV_FCzsPN_89OrkmJiKu5ruCcXxXZBMnqjl9zd6zMVWLa7o732UyRDUgf2cYhdZFXMluRf32A9bei67MDCgBOFJtpq_Pewm4BKNS8a-MRmNRdp9N5RLS1l-8nJaJsvMw8HDpo97N-_k4Vs_waYAEhS665H8sFniGtQjO65TZOxynOJqLpPt6aO2TNdPv_AczmrXXHRnpetNyc2Nk4n2q2KZR7ZPIPUVlF3GeEhf7jHcDTVM7gNY4kVEcUDUrkdUma69PGMSHleAqNSLhsFSf-O55nL6X0a9jIYOUhfXcLRx2vTWRJ_-TXoG2syQ3M2nyc4brckCBQCyjeXTcBhKMxrNvgd5zxGVvHCgUob7wZkFIB5ZEPUmPjY-zeiqsKQxDwflyJZuTn_AXU5pb7LxeAangkslAA2ubc8abhA1xMAXLAE6jtws71WLslGmQqiZW0o83AAKA91E-t5WV1DYkE6W-B01XhjUh5P8gxzr7gWXe9WtMkawmDKn5AaNpz4wYlDmpQLQs8HZqI0idfc1O23T-ZxNWFUfIfK-IwKYJX1r3yBKHll8iTxRUsAlpa3rA3-j3d8z4xejQXtp_X74BTnqjNva5rNy1uCTU-uwtgMgKXWf7iYQ-qFz08MEdEtI6AGdplifzr5f1HchE4tmCcRVIMmSxzXZRQqC8xYNzjQF3eTzAPMN6n4A4Wv9qP1Ahg4jD5_vmbihkFEmVJZMuCLsrbf11xIGss71Byqa9-IM1vltcPsLmFOsPnq03x1Wh0wTGiqW0hHYtdn-bVh4N1I6zNaAUsHGgcR0gd28G06HZjrpP6XZ2YQYVOSCzncseRJWDulhMDa3FcghPCslhsyXhPwsDmeTc8pA6jNonqxu6MtqKp_Gm0COMvqEwfnSiE-DHkUd98bFzPLDsIvFIAhIIfprLvRCxISKENqYWx5e-Bg5BmOjPUBhnzeq-fFlXh8T5YLJkYgmJDStMDpnD2cAb1FWY2KEVR14-Kwg32_2d9o6ud2NCKHHjrztOOJfxwD4ftaODI3a2tNM_dW2UtBYHry3-xoEjNTVALEZdTszaY3TBIjkrntwW3MjcdhAeUE9p1d7Lk5YsZ7_ZNk0SZ9GsNSZz87JnWz0NzU20-VqNqzE0BB-r15W4HGgwSJzafhrD5mgCaCgHb6lQRty2eaFghLuPhPF3jWAoNShKfu6qTeSDFiPW1YOiIT3gPOnEToX2RKuNM-LtTsGsSLnhZXcCjCBCsZdBG9oXwqInYYFi4-c7x5XT5idPnBzj9OQyK08MDOkEJifHwgqht5EtKZycJ0UdiMQGDkmZnUU_EHh5Mkc1HScyw9SCMQBzC9K8Ii9mEhy2IGxVltfpjLZsW_aSKGr1pyUsn9QEFKvkYItYSVb3FaQ6booURkxMp_4aGGb-apwts2R0mg2cg3w6NK86DDePTVTMQGmv8pxTQa7p9GlETkc7ZkZM0DwnzcIvJ5DHVDDMS_GUlhi1PncUakiCeOr18k3tKiv10Fz4OWXZ-FoFBrcFMsNlrypMYbbcmx-Xdq3hJ_cElC8YbASG88jYO3BMi107fpmAOvgInUFDRFrKKYBQH5T3mdKIfmph6shDofBUZ6JU_H7hgbCtw-j6dqcMG7GHGpwNxQ7V9XUQ-v59A5pBFxpnBGzzstdYiZdDfY9yl8yIKql2C-EiEOO2oVcPvnJ7hjfW-xnbE6MVMb_ciBVujZ4qijNYkAXmHgTzX5fPkEo67rHm1aUJbOi3roXVOG6xlR-DjIMPiS844z3nudz_HXl5fjCOw2PS7pevcNtS57YrwuVWXXFRXgjqdJBzbnD-nmwJOJOOyK68vgYYdIiZBLLHnevLi3ckwIKHFqa_hEgTH7-YswqYIYENYgK4v-wz-90rq7kCJMQDUgUH8ld3WiSOS4ZlaZvqLZnihc4D0aU4exbgsvgww8SWjcvhW8Ore4q0XqT6VGOnx9lTYLZ-MV1ye8-hLUqqvgh5ugG8ycrxbUv227Kt0nCFo36xsFI1aUNptjpOKfDmIB1D21roujRFMT6fQ187j_OqQSHf5rQPkb6YKzYBhhr9mNCL0i4eQLB6K69VNUQ6RtnJZ9Y-_PCRrYcFd-baxN4rKiiIK84I44GSGR3GsmWoKcowoLx1-4CNxB7yAPT7kyeESym4q_01Kv6YfWKh1XcSEz8F8RuS1SNxFHwuLdPhJ-H5tW_POqC4TPpBQtzz7w4L2ayGy1mDGlk4tFydNU0wF20uafE5FxtK88UXYwTvyHNRBoVzP2DI5NAkXDubWzdT5bAiZr3AGgcghMIQF-I-dHDWVfbWjQzRITPxe79cMZBeFIBdDLlarRv5u0Ji5Dy94K1fuSDLqV2wO431w8FPUcX-PXtbbA7CfyM4pAoKQp7PnbVifLH3DTqA1Yy0fQIHWgS77XAWQ8BRqQhhuj1yDgNSuQQVsnsZiybGW-Yb1gCw7cpr-Qabs4dIKCiFvs8kvr0LY7AkYuyY09TLJsHcPGK0AfS_EIqMMlncTZXR-e8FkqIWNAE4vKsr-UXBEDYweBz_tMlOuxRIRDH05RQLJBMcENqb5yWiak8uPNe3QI-_ANA4a98WSy6asSF4CkV_D5nGNtmUVpbpHMXvVHzBknORHhqHMQ0CYMEABf53Zec9po7arc_ewzHN1n3phP7UT_R0bSCqWpk0HILp1-HAan6x7TSk9u1qkInwEvxv3Uw_heOnhJoE7bF6Avslia3bpG87cxnfGoQ8J_wh29uqyaYsA7jEtPhIOQCR2CxGYsLV8m-YGyTvd2Sr7jX8s-aBBel1AbcDh6isXGOBlvr-s-eX9xd-b6dTTmToBnmulUTMgYVpjFSZz3bmS6ZPVkF4epb_rsOGcOMfPJzx1eNoCBMXBFU5GQAcVHxBFcJSJKeYqfW34w9BKiKWb_iiIMbq286N9abyOCJOSZbzxyTR8sK-vDHDp2NrMaUlzUviTAI53YbskBo743mFogTlJkbakJ1w7vdJCOJAAMCbJsUbnbFtvmX-kC0xdO5RLw4T__6lMfSHSmqeGC9yO0aM77eRvEusrrV-s3PnXtlRNgMYmLrkwfkcsYiWK7qbYPEqBneusVVELrbtq-1Sag8KU2Iw4ItY9r57GXzLfnGLeeARYrBRdhoobvg2e5FxMM0zJRFKR9M-22w3sCf11c_3VrhhwatkvQTuQkq6QXKBCUsof6i0d3yLFzlzqEfWG69uWyoUu_5sGzuOy1d1rfQopmFAC5NAMcDa5p9sbAW3yJpQQkghlgKsOIDfiJvvHyB1DvLlPwu0SOT3jdq9mMPoIfIeaY9v4e1BAIc0DFd80javYkyErwWtETo9CWxyVZFAlfN_5MQGK3tN0JChJ2QlmYX6DugXCrJd8sdGb4Yd2BWnkDX3AObobEu5zLBPXOp8cMG9D_g1lkzUnO4DD1_GN-bGUorxdKmVKx4YSgAiuyrYvsujYzRqhnto5wL8Oo6i4XWZW9K0TQ3dAFnt2EE95U0ERtt6sEqM0HU6sOdurt6WTI60zolQdv35aj-CKB8ckfPzi9-DfFhxnnzz6XV0K6Ncz5TqKBRIcslcq0f9C3bOugStTUKuQtD_X0xQSMn3WljtPByIXdA9SWNUVR2X8Vpa20QJbwqcW4SykYzWT-wYFH7SnxsxG4AcHLbJ-IA9rZCVKQgE1lg7JLcSStRCk9BhXb5K6nRx2BXmxbuZ1CCqK57FkKM_nCo8nxu5eoKs5cYIyQOBIVR6nHUjkhGG3NJSMR8rKVDQodg3ra575il_36BXow**b62446d3f8ab57f8c7787ba5743df22236cf8367969005f4103cf0e62e45683b*1GEez1igykAbHoOBZafOQVCVKzrc0MC6PC6YAP2ez64; Max-Age=86400; Path=/; Expires=Fri, 05 Jun 2020 16:12:59 GMT; HttpOnly; SameSite=Lax
```

The Cookies tab in the dev tools also notes that the cookie that it tries to set is "malformed". I'm not clear if malformed includes being too big, or whether there's something else wrong with it too. Either way, that seems to be the problem.

If I leave out the ID token and don't store that in the session, I can get logged in okay - or I can have the ID token but not the access token.

So, I'm now using the latest release - 0.13.0 - and I'm able to store the access token in the session/cookie. I don't currently need to store the ID token in the client session, but I'm sure that'd be useful to other folks - I guess it depends on the size of your ID and access tokens. In my case, I seem to be able to choose either the ID token or the access token to store in session state, but not both.

I'm not sure what the fix would be, or whether this ticket should be closed, but hopefully these findings are useful to someone in creating a fix. Please give me a shout if any more info is required.

**Comment by matthieuh** [comment]
I experienced the same issue discovered it since auth was working with short email address but not with long one x) Disabling IdToken storing fixed it on my side. I guess that depending if you are using accessToken or IdToken you can disable the other waiting to have a proper fix on versions > 0.7.0

**Comment by Widcket** [comment]
Hi everyone, can you please check if this issue still happens with the new [v1.0.0-beta.0](https://github.com/auth0/nextjs-auth0/releases/tag/v1.0.0-beta.0) release? We completely reworked the session code, and we now support cookie stitching (the length of the cookie should no longer cause issues). Your feedback is welcome.
